### PR TITLE
range input

### LINKED ISSUE #905: idea: slider / range behavior
This could be easily implemented with https://github.com/NightCatSama/vue-slider-component

### PR DESCRIPTION
**Pull request by nelsonpecora**
* closes #905
* allows single-point and dual-point ranges
* uses material design spec
* shows tooltips when focusing handles (if enabled)
* full keyboard support

**Note: Screenshots are not real world use cases**

Single point, no tooltips, no custom min/max labels

Single point, tooltips, custom min/max labels

Dual point, tooltips, custom max label

### CODE REVIEW

**Comment by rmfarrell** [comment]
This is really lovely. 

I'd just say that in general, you could kill a lot of complexity if these were two separate components: `range` and `slider`

File: inputs/range.vue [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+              if (val === max) {
+                return maxLabel;
+              }
+            }
+          }
+        }
+      });
+
+      // add keyboard support
+      _.each(findAll(el, '.noUi-handle'), (handle, index) => {
```
Comment (rmfarrell) [inline inputs/range.vue]: consider native JS here

File: inputs/range.vue [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+      slider.create(el, {
+        start: this.start,
+        step: step,
+        range: { min, max },
+        format: {
+          to(val) {
+            return step >= 1 ? parseInt(val) : val;
+          },
+          from(val) {
+            return step >= 1 ? parseInt(val) : val;
```
Comment (rmfarrell) [inline inputs/range.vue]: Can you get away with a simple typecast here? `Number(val)`

File: inputs/range.vue (hunk 1 above)
Comment (nelsonpecora) [reply inputs/range.vue]: NodeLists aren't arrays, so IE in particular doesn't have a native `forEach` method on their prototype 😄

**Comment by nelsonpecora** [comment]
I was considering having separate `slider` and `range` inputs, but that'd lead to a bunch of duplicated functionality and styles, and the terms themselves are interchangeable (the actual differentiation is how many points a slider/range has, which is not necessarily limited to 2, conceptually).

File: inputs/range.vue (hunk 1 above)
Comment (rmfarrell) [reply inputs/range.vue]: that's true for IE but you can call the prototype in the contexts of a NodeList, if we're using the Babel polyfill it adds this to NodeList and we don't support IE anyway.

**Comment by rmfarrell** [comment]
I'm still a 👍 for this, I just think we should consider mixins in the future. the shared methods can make use of the spread operator and you wouldn't need all these `if` statements. Having one vs. multiple sliders is where a lot of the logic tends to diverge.

File: inputs/range.vue (hunk 2 above)
Comment (nelsonpecora) [reply inputs/range.vue]: fixed!

File: inputs/range.vue (hunk 1 above)
Comment (zonika) [reply inputs/range.vue]: I've found that `Array.from(NodeList).forEach` works natively

File: package.json [hunk 3]
```diff
@@ -160,5 +161,7 @@
   "peerDependencies": {
     "amphora-search": "^6.2.0"
   },
-  "dependencies": {}
+  "dependencies": {
+    "nouislider": "^12.0.0"
```
Comment (zonika) [inline package.json]: this makes me sad!!!!!!!!

Comment (nelsonpecora) [reply package.json]: wait why

Comment (nelsonpecora) [reply package.json]: (nouislider is a really good, pretty lightweight library that implements a lot of functionality that is just completely unsupported with native html sliders)

Comment (nelsonpecora) [reply package.json]: oh it doesn't need to be a regular dep, right. good catch!!!

Comment (zonika) [reply package.json]: i liked having 0 deps!!!!!!!

**Pull request merged**